OneOneOne/reciever/tests/test_message_checks.py

```diff
@@ -0,0 +1,150 @@
+import unittest
+from pathlib import Path
+from builder.pystache_message_builder import PystacheMessageBuilder
+import xml.etree.ElementTree as ET
+from definitions import ROOT_DIR
+from reciever.message_checks.checks import *
+
+
+class MessageCheckTests(unittest.TestCase):
+    inputXmlFileDir = Path(ROOT_DIR) / 'reciever' / 'tests' / 'input_xmls'
+    builder = PystacheMessageBuilder(str(inputXmlFileDir), 'base_input')
+
+    def test_action_not_matching_service(self):
+
+        with self.subTest("Two differing services result in a 500 error"):
+            service_dict = {'action': "urn:nhs-itk:services:201005:SendNHS111Report-v2-0-ThisDoesNotMatchBelow",
+                            'service': "urn:nhs-itk:services:201005:SendNHS111Report-Bad_Service-ThisDoesNotMatchAbove"}
+
+            msg = self.builder.build_message(service_dict)
+            message_tree = ET.fromstring(msg)
+            mv = CheckActionTypes(message_tree)
```

Comment:
it looks like lines 19 and 20 are quite common here with the other tests in this file. Are these lines just setting up message_tree the check needs? Perhaps a helper function here like `generate_message_tree` with the `service_dict` passed in. Then this could be inlined as `mv = CheckActionTypes(generate_message_tree(service_dict))` might make these bit easier to read. Also `mv` might better be renamed to something else

Reply:
Killed two birds with one stone and inclined the whole thing:
`fail_flag, response = CheckActionTypes(self.generate_message_tree(service_dict)).check()`

Reply:
I've also updated some of the names from the `test_message_handler` as well